src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/gadgets/MultiTool.java

```diff
@@ -73,25 +78,49 @@ protected ItemUseHandler getItemUseHandler() {
         return e -> {
             Player p = e.getPlayer();
             ItemStack item = e.getItem();
+            PersistentDataContainer container = item.getItemMeta().getPersistentDataContainer();
             e.cancel();
-
-            int index = selectedMode.getOrDefault(p.getUniqueId(), 0);
-
-            if (!p.isSneaking()) {
-                if (removeItemCharge(item, COST)) {
-                    SlimefunItem sfItem = modes.get(index).getItem();
-
-                    if (sfItem != null) {
-                        sfItem.callItemHandler(ItemUseHandler.class, handler -> handler.onRightClick(e));
+            
+            int index = 0;
+            if (container.has(key, PersistentDataType.STRING)) {
+                for (int i = 0; i < modes.size(); i++) {
+                    if (modes.get(i).getItemId().equals(container.get(key, PersistentDataType.STRING))) {
```

Comment:
I thought this MultiToolMode would be an enum _sigh_, ok so it's an object which we use very awkwardly right now...

What we can do but I don't like is to instead have a getId() getter, this is gonna be the index so we can just do `modes.get(getId())`

We should have an actual ID here, I won't put that onto you now though

Reply:
I have seen you update the branch many times but didnt adress this commnet

Reply:
Sorry, seems like I forgot about this one. From what I remember, SlimefunItem already has a `getId()` getter and Multitool class inherits it. I tried to find a way to create such a getter under a different name for it, and I have no idea what it would actually do. Replace the _index_? If so, the code above uses _meta_ variable, which is provided by an event - I would need to pass it as a variable to that getter (so basically I would have to do `getIndex(meta)` or something like that) and it feels redundant. Unless I'm missing something here, I think it would be better to leave it as it is for now and refactor it in the future.

If it changes anything, it uses a filter() method now instead of a for loop, like that:
`int index = !PersistentDataAPI.hasString(meta, key) ? 0 :
                IntStream.range(0, modes.size()).
                filter(i -> modes.get(i).getItemId().equals(
                    PersistentDataAPI.getString(meta, key, modes.get(0).getItemId())
                )).findFirst()
                .orElse(0);
`

Sorry if I missed something crucial or obvious here, I don't have much experience with Java, let alone Slimefun.